python/requirements.txt

```diff
@@ -0,0 +1,4 @@
+numpy==1.16.4
+scipy==1.2.2
```

Comment:
Heh, I didn't realize we directly use scipy. OK.

Out of ignorance, what effect does this have on users of pyspark? like, without this, they are not guaranteed to have the right supporting dependencies right now?

Reply:
Oh, so this does nothing unless you manually pip install from this file, and it's there for devs? OK.

Reply:
According to the UT, something doesn't work without these. 
The file itself is just a conventional file for `pip`. Users can setup easily with `pip install -r python/requirements.txt`.

If we use `==`, all mismatched versions (both lower and higher) will be removed and reinstalled with the exact versions.

Reply:
For `scipy`, I overlooked the real test cases. Yes. It was just a test dependency like the following.
```
Skipped tests in pyspark.mllib.tests.test_linalg with python:
    test_classification (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_clustering (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_convert_to_vector (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_dot (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_regression (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_serialize (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'
    test_squared_distance (pyspark.mllib.tests.test_linalg.SciPyTests) ... skipped 'SciPy not installed'

Skipped tests in pyspark.tests.test_serializers with python:
    test_serialize (pyspark.tests.test_serializers.SciPyTests) ... skipped 'SciPy not installed'
```

Since this is moved to `dev`, I'll update the PR description and keep this for `dev` purpose~